### PR TITLE
Apple Notes: add compatibility with older unsupported versions of macOS

### LINKED ISSUE #159: Apple Notes Importer doesn't import anything
I installed the Importer plugin today to try to import my apple notes, however nothing is imported and there is a console error (although no user-visible error):

Screen Shot 2023-10-26 at 8 09 22 PM

Here's the full error:
```
Uncaught (in promise) Error: Error: mode should be one of: ascii column csv html insert line list quote tabs tcl

    at Socket.h (plugin:obsidian-importer:71:260)
    at Object.onceWrapper (node:events:628:26)
    at Socket.emit (node:events:513:28)
    at addChunk (node:internal/streams/readable:324:12)
    at readableAddChunk (node:internal/streams/readable:297:9)
    at Readable.push (node:internal/streams/readable:234:10)
    at Pipe.onStreamRead (node:internal/stream_base_commons:190:23)
h @ plugin:obsidian-importer:71
onceWrapper @ node:events:628
emit @ node:events:513
addChunk @ node:internal/streams/readable:324
readableAddChunk @ node:internal/streams/readable:297
Readable.push @ node:internal/streams/readable:234
onStreamRead @ node:internal/stream_base_commons:190
```

And here's my debug info:
SYSTEM INFO:
	Obsidian version: v1.4.16
	Installer version: v1.4.13
	Operating system: Darwin Kernel Version 20.6.0: Thu Jul  6 22:12:47 PDT 2023; root:xnu-7195.141.49.702.12~1/RELEASE_X86_64 20.6.0
	Login status: logged in
	Catalyst license: supporter
	Insider build toggle: off
	Live preview: on
	Legacy editor: off
	Base theme: dark
	Community theme: none
	Snippets enabled: 0
	Restricted mode: off
	Plugins installed: 1
	Plugins enabled: 1
		1: Importer v1.5.3

Note: this is on a fresh vault with nothing done except installing the importer plugin.

### PR DESCRIPTION
**Pull request by mirnovov**
Closes #159 (at least the remainder of it that wasn't fixed in 1.5.4). 

Currently, SQLite transmits data to the Apple Notes importer via JSON. However, on older versions of macOS, JSON is not supported, so the importer fails on these. This PR makes it so it checks whether JSON is supported, and if not, it instead uses the SQL literal mode (which the importer then converts into JS objects manually). 

Note that this PR may not in fact be desirable to merge as this adds additional maintenance burden for versions that will eventually not be supported by Electron anyway, and further diverges the vendored SQLite dependency from the original version. On the other hand, it would be nice to support older OS versions and it's probably what users expect.

### CODE REVIEW

File: src/formats/apple-notes/sqlite/index.js [hunk 1]
```diff
@@ -1,12 +1,14 @@
 import { Platform } from 'obsidian';
 import { os, path } from '../../../filesystem';
+import { supportsJson, sqlToJson } from './fallback';
 import { error, raw, sql } from './utils.js';
 
-const crypto = Platform.isDesktopApp ? window.require('node:crypto') : null;
-const child_process = Platform.isDesktopApp ? window.require('node:child_process') : null;
+export const crypto = Platform.isDesktopApp ? window.require('node:crypto') : null;
```
Comment (tgrosinger) [inline src/formats/apple-notes/sqlite/index.js:6]: Why do these need to be exported?

File: src/formats/apple-notes/sqlite/fallback.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+	
+	const columnNames = [];
+	let i = 0;
+	let columnPos = -1;
+	let row = {};
+	
+	while (i < sql.length) {
+		let token = '';
+		
+		if (sql[i] == '\'') {
```
Comment (tgrosinger) [inline src/formats/apple-notes/sqlite/fallback.js:22]:
```suggestion
		if (sql[i] === '\'') {
```

File: src/formats/apple-notes/sqlite/fallback.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+			// String/hex-encoded blob
+			i++;
+						
+			while (i < sql.length) {
+				if (sql[i] != '\'') {
```
Comment (tgrosinger) [inline src/formats/apple-notes/sqlite/fallback.js:27]:
```suggestion
				if (sql[i] !== '\'') {
```

File: src/formats/apple-notes/sqlite/fallback.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+					// Not quote
+					const offset = sql.indexOf('\'', i);
+					token += sql.substring(i, offset);
+					i = offset;
+				}
+				else if (sql[i + 1] == '\'') {
```
Comment (tgrosinger) [inline src/formats/apple-notes/sqlite/fallback.js:33]:
```suggestion
				else if (sql[i + 1] === '\'') {
```

File: src/formats/apple-notes/sqlite/fallback.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+					i += 2;
+				}
+				else {
+					// Closing quote
+					i++;
+					break;
+				}
+			}
+		}
+		else if (sql[i] == 'N') {
```
Comment (tgrosinger) [inline src/formats/apple-notes/sqlite/fallback.js:45]:
```suggestion
		else if (sql[i] === 'N') { 
```

**Review by tgrosinger — APPROVED** [review]
Looks great, thank you.

File: src/formats/apple-notes/sqlite/index.js (hunk 1 above)
Comment (mirnovov) [reply src/formats/apple-notes/sqlite/index.js:6]: At least one is used in the new ‘fallback.js’ file.

**Pull request merged**